docker/Dockerfile

```diff
@@ -1,12 +1,23 @@
-FROM golang:alpine AS builder
-
-RUN mkdir /src
-ADD . /src
-
-RUN apk add --no-cache make git curl && \
-	cd /src && \
-	mkdir -p output/bin && \
-	make V=1 bin/step
+FROM --platform=$BUILDPLATFORM golang:alpine AS builder
+
+WORKDIR /src
+COPY go.mod go.sum .
+RUN go mod download
+COPY . .
+RUN apk add --no-cache make git curl
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETVARIANT
+RUN --mount=type=cache,target=/root/.cache/go-build \
```

Comment:
These cache mounts really only improve performance on iterative builds of the same image. They won't speed up the release.